crates/sqlexec/src/planner/physical_plan/copy_to.rs

```diff
@@ -124,6 +129,35 @@ impl CopyToExec {
                 };
                 get_sink_for_obj(format, &access, &s3_options.location)?
             }
+            (CopyToDestinationOptions::Azure(azure_options), format) => {
+                // Create storage options using well-known key names.
+                let opts = StorageOptions::new_from_iter([
+                    (AzureConfigKey::AccountName.as_ref(), azure_options.account),
+                    (AzureConfigKey::AccessKey.as_ref(), azure_options.access_key),
+                ]);
+                let access =
+                    GenericStoreAccess::new_from_location_and_opts(&azure_options.location, opts)
+                        .map_err(|e| DataFusionError::External(Box::new(e)))?;
+
+                // TODO: It's weird we need to do this here, but
+                // `get_sink_for_obj` is expected a path relative to the root of
+                // the store. The location we have here is the full url
+                // (azure://...) and so will actually cause object store to
+                // error.
+                //
+                // By converting to a data source url, we can get the path we
+                // need.
+                //
+                // @vaibhav I'd like for us to look into switchin all object
+                // store "locations" to use the full url (with scheme) so that
+                // we can be consistent with this. It would also help with user
+                // experience since they wouldn't need to know which part of the
+                // location is the "bucket" and which is the "location" (path).
+                let source_url = DatasourceUrl::try_new(&azure_options.location)
+                    .map_err(|e| DataFusionError::External(Box::new(e)))?;
```

Comment:
@vrongmeal I want to point this out that azure is being handled slightly differently than the other object stores (since we're going through the generic access). Instead of having separate "bucket" and "location" fields, we just have a single "location" field that has the full url to the object.

While this isn't a change a I wanted to make in this pr, I do think moving in that direction for the other stores would provide a better ux. Instead of the user needing to give us separate locations and buckets, they just throw the full url our way and we handle that directly.